src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1237,6 +1237,17 @@ class VectorStoreMaskNode : public VectorNode {
   static VectorStoreMaskNode* make(PhaseGVN& gvn, Node* in, BasicType in_type, uint num_elem);
 };
 
+class VectorMaskCastNode : public VectorNode {
```

Comment:
VectorMaskReinterpret seems better choice, since its a re-interpretation and not a casting (up/down).

Reply:
Considering masks have platform-specific representation, full-blown casts between different element types look more appropriate here.

In this particular case, the focus is on the cheapest possible case when representations share the same bit pattern and the cast degenerates into a no-op. But in the longer term, it makes perfect sense to support the full matrix of conversions and don't rely on `VectorLoadMask <=> VectorStoreMask` and intermediate canonical vector representation.

Reply:
Got it.